cmd/podman/machine/init.go

```diff
@@ -89,6 +88,10 @@ func init() {
 	)
 	_ = flags.MarkHidden("reexec")
 
+	UsernameFlagName := "username"
+	flags.StringVar(&initOpts.Username, UsernameFlagName, cfg.Machine.User, "Username used in qcow image")
```

Comment:
Should this be `cfg.Config.Machine.User` instead of `cfg.Machine.User`, given that's what the removed bit used?

Reply:
I tested this with cherry-pick (w/o additional alterations). And it resulted with
`"RemoteUsername": "user",` in json config when called w/o explicit value and with `"RemoteUsername": "core",` giving `--username core` in the command line. I will investigate the code to understand better how it works. Wrote it this way because lines below references config settings with `cfg.Machine.*` - w/o additional `.Config`. And it was the only line with `cfg.Config`.

I will take a more detailed look before implementing this change. Keeping the draft for now.

Reply:
If I read this line correctly: https://github.com/containers/podman/blob/bec7e8a6456de2853e5a5f733d144dffc9885073/pkg/domain/entities/engine.go#L33

Then Config is just expanded/blended into PodmanConfig, so, we can use it bypassing one dereference. My knowledge of advanced Go features is still limited. But at least VS Code code navigation navigates to the same place with `.cfg.Machine` and `.cfg.Config.Machine`. And, as I mentioned above, this worked in build the same way I described here, so, I guess this is correct.

@mheon Do you still insist on changing this (and only this) line?